### PR TITLE
🚀 Add `world with name` expression

### PR DESCRIPTION
**Pull request by AyhamAl-Ali**
### Description

This PR adds a new expression to get the world from string, this also matches the literal syntax to fix any related issues

**NOTE**: This could be similar to #4408 but not the same.

---
**Target Minecraft Versions:** Any
**Requirements:** None
**Related Issues:**  [#3971 comment](https://github.com/SkriptLang/Skript/issues/3971#issuecomment-836403090) #1144

### CODE REVIEW

**Comment by AyhamAl-Ali** [comment]
I will look into the build error as soon as I can.

**Comment by TPGamesNL** [comment]
I think the syntax should be the same as world's literal parsing, as that would also fix https://github.com/SkriptLang/Skript/issues/1144

**Comment by AyhamAl-Ali** [comment]
> I think the syntax should be the same as world's literal parsing, as that would also fix #1144

Sure, I will push the changes and do some more tests tonight just to make sure of everything

File: src/main/java/ch/njol/skript/expressions/ExprWorldFromName.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+@Name("World From Name")
+@Description("Returns the world from a string.")
+@Examples({"world from the name {game::world-name}",
+			"the world \"world\""})
+@Since("INSERT VERSION")
+public class ExprWorldFromName extends SimpleExpression<World> {
+
+	static {
+		Skript.registerExpression(ExprWorldFromName.class, World.class, ExpressionType.SIMPLE, "[the] world [(of|from|with)] [the] [name] %string%");
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprWorldFromName.java:44]: I'd prefer this pattern, as also seen in [ExprNamed](https://skriptlang.github.io/Skript/expressions.html#ExprNamed):
```suggestion
		Skript.registerExpression(ExprWorldFromName.class, World.class, ExpressionType.SIMPLE, "[the] world (named|with name) %string%");
```

File: src/main/java/ch/njol/skript/expressions/ExprWorldFromName.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("World From Name")
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprWorldFromName.java:36]:
```suggestion
@Name("World from Name")
```

File: src/main/java/ch/njol/skript/expressions/ExprWorldFromName.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	static {
+		Skript.registerExpression(ExprWorldFromName.class, World.class, ExpressionType.SIMPLE, "[the] world [(of|from|with)] [the] [name] %string%");
+	}
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<String> worldName;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprWorldFromName.java:52]:
```suggestion
	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
```

File: src/main/java/ch/njol/skript/expressions/ExprWorldFromName.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	@Override
+	@Nullable
+	protected World[] get(Event e) {
+		String worldName = this.worldName.getSingle(e);
+		if (worldName == null)
+			return null;
+
+		return new World[] {Bukkit.getWorld(worldName)};
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprWorldFromName.java:64]: This method may not return an array with null values

File: src/main/java/ch/njol/skript/expressions/ExprWorldFromName.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	@Override
+	public Class<World> getReturnType() {
+		return World.class;
+	}
+
+	@Override
+	public String toString(@Nullable Event e, boolean debug) {
+		return "the world of name" + worldName.toString(e, debug);
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprWorldFromName.java:79]:
```suggestion
		return "the world of name " + worldName.toString(e, debug);
```
Should also be adjusted for syntax changes

**Review by TPGamesNL — CHANGES_REQUESTED** [review]

**Review by TPGamesNL — APPROVED** [review]

**Pull request merged**